avm/res/machine-learning-services/workspace/main.bicep

```diff
@@ -0,0 +1,477 @@
+      }
+    } : null
+    imageBuildCompute: imageBuildCompute
+    primaryUserAssignedIdentity: primaryUserAssignedIdentity
+    publicNetworkAccess: !empty(publicNetworkAccess) ? any(publicNetworkAccess) : (!empty(privateEndpoints) ? 'Disabled' : 'Enabled')
+    serviceManagedResourcesSettings: serviceManagedResourcesSettings
+    sharedPrivateLinkResources: sharedPrivateLinkResources // Note: This property is not idempotent. Neither with [] or `null`
+  }
+}
+
+module workspace_computes 'compute/main.bicep' = [for compute in (computes ?? []): {
+  name: '${workspace.name}-${compute.name}-compute'
+  params: {
+    machineLearningWorkspaceName: workspace.name
+    name: compute.name
+    location: compute.location
+    sku: compute.?sku
+    managedIdentities: compute.?managedIdentities
+    tags: compute.?tags
+    deployCompute: compute.?deployCompute
+    computeLocation: compute.?computeLocation
+    description: compute.?description
+    disableLocalAuth: compute.?disableLocalAuth
+    resourceId: compute.?resourceId
+    computeType: compute.computeType
+    properties: compute.?properties
+  }
+  dependsOn: [
+    workspace_privateEndpoints
+  ]
+}]
+
+resource workspace_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: workspace
+}
+
+resource workspace_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for (diagnosticSetting, index) in (diagnosticSettings ?? []): {
+  name: diagnosticSetting.?name ?? '${name}-diagnosticSettings'
+  properties: {
+    storageAccountId: diagnosticSetting.?storageAccountResourceId
+    workspaceId: diagnosticSetting.?workspaceResourceId
+    eventHubAuthorizationRuleId: diagnosticSetting.?eventHubAuthorizationRuleResourceId
+    eventHubName: diagnosticSetting.?eventHubName
+    metrics: [for group in (diagnosticSetting.?metricCategories ?? [ { category: 'AllMetrics' } ]): {
+      category: group.category
+      enabled: group.?enabled ?? true
+      timeGrain: null
+    }]
+    logs: [for group in (diagnosticSetting.?logCategoriesAndGroups ?? [ { categoryGroup: 'allLogs' } ]): {
+      categoryGroup: group.?categoryGroup
+      category: group.?category
+      enabled: group.?enabled ?? true
+    }]
+    marketplacePartnerId: diagnosticSetting.?marketplacePartnerResourceId
+    logAnalyticsDestinationType: diagnosticSetting.?logAnalyticsDestinationType
+  }
+  scope: workspace
+}]
+
+module workspace_privateEndpoints 'br/public:avm/res/network/private-endpoint:0.4.0' = [for (privateEndpoint, index) in (privateEndpoints ?? []): {
```

Comment:
The [PE schema update](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#private-endpoints) was merged. Would be great if we could apply it here.

Reply:
Done: [9f771a9](https://github.com/Azure/bicep-registry-modules/pull/1055/commits/9f771a9462ed9130ac3f7130e6c7ddd0755437a9)